sample-app/src/hooks/usePage.tsx

```diff
@@ -0,0 +1,33 @@
+import { SearchIntent } from "@yext/answers-headless";
+import { useLayoutEffect } from "react";
+import { useAnswersActions } from "@yext/answers-headless-react";
+import { executeSearch, getSearchIntents, updateLocationIfNeeded } from "../utils/search-operations";
+
+/**
+ * Sets up the state for a page
+ * @param verticalKey - The verticalKey associated with the page, or undefined for universal pages
+ */
+export default function usePage(verticalKey?: string) {
+  const answersActions = useAnswersActions();
+  useLayoutEffect(() => {
+    const stateToClear = verticalKey
+      ? { universal: {} }
+      : { vertical: {} };
+    answersActions.setState({
```

Comment:
should we add a reset on facets and filters for this page setup? I noticed sometimes query may change between pages, but navigating back to previous pages with filters/facets, the selected filters/facets may not correspond with the new query. Unless we want to leave this behavior up to user?

Reply:
Ya, I noticed that as well. I'll update to clear filters here as well